diff --git a/src/modules/hyprland/workspace.cpp b/src/modules/hyprland/workspace.cpp
index 4cdd8910..e9d4d1e0 100644
--- a/src/modules/hyprland/workspace.cpp
+++ b/src/modules/hyprland/workspace.cpp
@@ -73,7 +73,8 @@ bool Workspace::handleClicked(GdkEventButton* bt) const {
         if (m_workspaceManager.moveToMonitor()) {
           m_ipc.getSocket1Reply("dispatch focusworkspaceoncurrentmonitor " + std::to_string(id()));
         } else {
-          m_ipc.getSocket1Reply("dispatch workspace " + std::to_string(id()));
+          std::string cmd = fmt::format("~/.config/hypr/scripts/workspace.sh workspace {}", id());
+          util::command::execNoRead(cmd);
         }
       } else if (!isSpecial()) {  // named (this includes persistent)
         if (m_workspaceManager.moveToMonitor()) {
